src/bst.py

```diff
@@ -94,15 +94,15 @@ def post_order(self, root='root'):
             for node in root.children():
                 for node in self.post_order(root=node):
                     yield node
-            yield root
+            yield root.val
 
     def breadth_first(self, start):
```

Comment:
Here, you should have a default value for start (like None), and then set it to self.root if necessary in your method